### PR TITLE
External gRPC Cloud Provider

### PR DESCRIPTION
**Pull request by dbonfigli**
#### Which component this PR applies to?

cluster-autoscaler

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Add a cloud provider to allow external implementations as described in the proposal #3140 [Pluggable Cloud Provider over gRPC](https://github.com/kubernetes/autoscaler/blob/68c984472acce69cba89d96d724d25b3c78fc4a0/cluster-autoscaler/proposals/plugable-provider-grpc.md)

#### Does this PR introduce a user-facing change?

```release-note
New cloud provider: external gRPC
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by mwielgus — CHANGES_REQUESTED** [review]

**Comment by mwielgus** [comment]
Please add OWNERS file (possibly with commented-out names) so we know who to contact in case of bugs, issues, reviews etc.

File: cluster-autoscaler/cloudprovider/externalgrpc/examples/external-grpc-cloud-provider-service/main.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```
[the remark this answers is not in the document]
Comment (dbonfigli) [reply cluster-autoscaler/cloudprovider/externalgrpc/examples/external-grpc-cloud-provider-service/main.go]: Updated.

**Comment by dbonfigli** [comment]
> Please add OWNERS file (possibly with commented-out names) so we know who to contact in case of bugs, issues, reviews etc.

Added.

**Review by mwielgus — APPROVED** [review]
/lgtm
/approve

**Pull request merged**